src/object/player.cpp

```diff
@@ -745,12 +745,13 @@ Player::update(float dt_sec)
     }
   }
 
-  if (m_does_buttjump)
+  if (m_does_buttjump || (m_stone && m_physic.get_velocity_y() > 30.f))
   {
     Rectf downbox = get_bbox().grown(-1.f);
     downbox.set_bottom(get_bbox().get_bottom() + 16.f);
     for (auto& brick : Sector::get().get_objects_by_type<Brick>()) {
-      if (downbox.contains(brick.get_bbox()) && brick.get_class_name() != "heavy-brick") {
+      // stoneform breaks through any kind of bricks
+      if (downbox.contains(brick.get_bbox()) && (m_stone || brick.get_class_name() != "heavy-brick")) {
```

Comment:
Not related to a change of yours, but it would be nice if `brick.get_class_name() != "heavy-brick"` is replaced with a `dynamic_cast`.

Reply:
Probably won't make much difference there but [string comparsion can be faster than dynamic_cast](https://stackoverflow.com/a/9778220).

Reply:
We should keep it consistent throughout the code. Other places all use `dynamic_cast`.